### PR TITLE
Build CoreCLR tests as platform independent (AnyCPU)

### PR DESCRIPTION
**Pull request by gbalykov**
After this change same test dll can be launched on different architectures.

Related issue: #31793 

cc @alpencolt @jkotas

### CODE REVIEW

File: src/coreclr/tests/dir.sdkbuild.props [hunk 1]
```diff
@@ -9,7 +9,7 @@
     <TargetFramework>$(NetCoreAppCurrent)</TargetFramework>
     <AppendTargetFrameworkToOutputPath>false</AppendTargetFrameworkToOutputPath>
     <EnableDefaultItems>false</EnableDefaultItems>
-    <Platform>$(BuildArch)</Platform>
+    <Platform>AnyCPU</Platform>
```
Comment (am11) [inline src/coreclr/tests/dir.sdkbuild.props]: Was this change required because `BuildArch` has different value in this case?

Comment (gbalykov) [reply src/coreclr/tests/dir.sdkbuild.props]: `BuildArch` is equal to actual arch specified for build (i.e. `arm`, `x64`)

**Comment by jkotas** [comment]
cc @dotnet/runtime-infrastructure

File: src/coreclr/tests/dir.sdkbuild.props (hunk 1 above)
Comment (jkotas) [reply src/coreclr/tests/dir.sdkbuild.props]: Can this line be deleted?

Comment (gbalykov) [reply src/coreclr/tests/dir.sdkbuild.props]: The same error with incorrect format appears if this line is removed

**Comment by jaredpar** [comment]
@BruceForstall you may want to run the JIT jobs on this change to verify they still function. I'm not sure of the appropriate AzDO legs to run here or would initiate them myself.

**Comment by gbalykov** [comment]
Is this change ok for merge?

**Comment by jaredpar** [comment]
@gbalykov need to get the build passing first. Re-kicked off the one failed leg (was hitting the NuGet restore bug).

I will assign someone out from infra to review today.

**Comment by BruceForstall** [comment]
I manually triggered an outerloop job, to ensure we have full pri-1 testing.

**Comment by jashook** [comment]
It is important to note that this change only allows us to run tests with no native components on different platforms that were built on.

**Comment by BruceForstall** [comment]
outerloop failed due to https://github.com/dotnet/runtime/issues/33047. @gbalykov after review, can you rebase and push so a manual run of outerloop will pick up https://github.com/dotnet/runtime/pull/33018?

**Review by trylek — APPROVED** [review]
LGTM modulo Bruce's suggestion. Further work is necessary to fully utilize this change for reusing managed test artifacts but I don't think it's blocking as this change by itself doesn't actually change the managed test reuse policy in our pipelines. When running tests locally, test builds are stratified into subfolders by build architecture that is moreover typically determined by our devbox so I don't think that relaxing the MSIL assembly headers to be tagged as AnyCPU poses any actual risk for our current testing.

**Comment by gbalykov** [comment]
@BruceForstall I've rebased

**Comment by BruceForstall** [comment]
I triggered an outerloop run.

**Comment by BruceForstall** [comment]
outerloop passed. crossgen-comparison failure is infra. merging.
